android/KMEA/app/src/main/java/com/tavultesoft/kmea/logic/ResourcesUpdateTool.java

```diff
@@ -269,6 +282,83 @@ private void addOpenUpdate(Bundle theResourceBundle) {
       addOpenUpdate(createKeyboardId(langid, kbid), null, theResourceBundle);
     }
   }
+
+  /**
+   * Check shared preference to see if an update notification should be ignored.
+   * The window is MONTHS_TO_IGNORE_NOTIFICATION from the last time the notification was ignored.
+   * @param id keyboard or lexical model ID
+   * @return true if the notification should be ignored
+   */
+  private boolean shouldIgnoreNotification(String id) {
+    SharedPreferences prefs = currentContext.getSharedPreferences(currentContext.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
+    SharedPreferences.Editor editor = prefs.edit();
+    String ignoredNotificationsStr = prefs.getString(PREF_KEY_IGNORE_NOTIFICATIONS, null);
+
+    /*
+     * Preference is a JSON Object (as a string)
+     * { PREF_KEY_IGNORE_NOTIFICATIONS :
+     *   { id1 : time1 ignored,
+     *     id2 : time2 ignored
+     *   }
+     * }
+     */
+    JSONObject ignoredNotificationsObj;
+    if (ignoredNotificationsStr != null) {
+      try {
+        ignoredNotificationsObj = new JSONObject(ignoredNotificationsStr);
+
+        Long lastIgnoredTime = ignoredNotificationsObj.optLong(id, 0);
+        if (lastIgnoredTime > 0) {
+          Calendar now = Calendar.getInstance();
+          Calendar ignoreUntilTime = Calendar.getInstance();
+          ignoreUntilTime.setTime(new Date(lastIgnoredTime));
+          ignoreUntilTime.add(Calendar.MONTH, MONTHS_TO_IGNORE_NOTIFICATION);
+          if (now.compareTo(ignoreUntilTime) < 0) {
+            return true;
+          }
+        }
+      } catch (JSONException e) {
+        Log.e(TAG, "JSON Exception parsing ignoreNotifications preference");
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Update preference to ignore notifications for keyboard / lexical model ID
+   * @param id : keyboard or lexical model ID to ignore for MONTHS_TO_IGNORE_NOTIFICATION months
+   */
+  private void setPrefKeyIgnoreNotifications(String id) {
+    SharedPreferences prefs = currentContext.getSharedPreferences(currentContext.getString(R.string.kma_prefs_name), Context.MODE_PRIVATE);
```

Comment:
Should the load of the editor be done inside the try/except and for the shortest time possible -- right before `putString`?

Reply:
Also please wrap this line :)

Reply:
> Should the load of the editor be done inside the try/except?

I assign pre-existing preferences  to `ignoredNotificationsStr` in line 295, and that gets checked before the try/except.

I'll wrap the line though.

Reply:
Ok, but the editor instantiated in 334 is not used until line 355. That's the line that should be moved, I think. :)